core/src/main/scala/kafka/server/KafkaConfig.scala

```diff
@@ -1959,10 +1961,18 @@ class KafkaConfig private(doLog: Boolean, val props: java.util.Map[_, _], dynami
     }
   }
 
-  def listenerSecurityProtocolMap: Map[ListenerName, SecurityProtocol] = {
-    getMap(KafkaConfig.ListenerSecurityProtocolMapProp, getString(KafkaConfig.ListenerSecurityProtocolMapProp))
+  def effectiveListenerSecurityProtocolMap: Map[ListenerName, SecurityProtocol] = {
+    val mapValue = getMap(KafkaConfig.ListenerSecurityProtocolMapProp, getString(KafkaConfig.ListenerSecurityProtocolMapProp))
       .map { case (listenerName, protocolName) =>
-      ListenerName.normalised(listenerName) -> getSecurityProtocol(protocolName, KafkaConfig.ListenerSecurityProtocolMapProp)
+        ListenerName.normalised(listenerName) -> getSecurityProtocol(protocolName, KafkaConfig.ListenerSecurityProtocolMapProp)
+      }
+    if (usesSelfManagedQuorum && !originals.containsKey(ListenerSecurityProtocolMapProp)) {
+      // Nothing was specified explicitly, so we are using the default value; therefore, since we are using KRaft,
+      // add the PLAINTEXT mappings for all controller listener names that are not security protocols
+      mapValue ++ controllerListenerNames.filter(cln => cln.nonEmpty && !SecurityProtocol.values().exists(_.name.equals(cln))).map(
```

Comment:
Suppose that the user has the following configuration:

```
listeners=SSL://9092,CONTROLLER://9093
controller.listener.names=CONTROLLER
```

With the change here, would we still default the CONTROLLER listener to PLAINTEXT? I can't shake the feeling that we would be better off forcing the user to be explicit about their intent. Perhaps in the one case when all of the other listeners are PLAINTEXT (as would be the case with quickstarts), then it is reasonable to also assume PLAINTEXT. But I'm inclined to be stricter if other security protocols are in use. What do you think?

Reply:
> I'm inclined to be stricter if other security protocols are in use. What do you think?

Yeah, I agree that we should only map when SSL and SASL_{SSL,PLAINTEXT} are not in use.  I've pushed a commit that implement this.